### ISSUE TITLE
GD-427: Typecast all variables to avoid errors when script errors are enabled for the plugin

### ISSUE THREAD

**Issue by rafaelGuerreiro** [MikeSchulze/gdUnit4#427]
**Describe What**
Add type information in all declarations in gdUnit4

**Describe Why**
I'm developing an addon and I'd like to test it, but because I have Untyped Declarations set to Error, I can't run gdUnit4.

![image](https://github.com/MikeSchulze/gdUnit4/assets/347037/8903480c-ef95-4995-8ca6-7f0bc21f3eb9)

**Dev hints**
I can implement this issue, but because this project is quite large, I'd like to know if this is desirable from the team.

**Comment by MikeSchulze** [comment]
I know this is the case, but at the moment you should not enable the integration of plugins for the detection of warnings and errors.

To fix all the possible errors and warnings will be a bigger task. These code is mainly migrated from Godot 3 where was untyped.
Unfortunately, Godot does not allow you to set this specifically for a particular plugin, so you currently have no choice but to work with the exclusion of plugins.

**Comment by rafaelGuerreiro** [comment]
You're right. But we can mostly use Variant as a placeholder and Godot will be happy about it.

Would it be okay if I attempted this?

**Comment by MikeSchulze** [comment]
No! Don't use Variant instead.
I actually have started code cleanup and the type should be correct typed, when do code cleanup the code should be better readable as before ;)

**Comment by MikeSchulze** [comment]
https://github.com/MikeSchulze/gdUnit4/pull/438

**Comment by MikeSchulze** [comment]
@rafaelGuerreiro I finally made it.

Feel free to pull the master version, GdUnit4 v4.3.0.
Please note, the version 4.3.0 is now requiring Godot 4.2 as minimum version.

Please give me feedback.

**Comment by rafaelGuerreiro** [comment]
@MikeSchulze, it's working fine so far, thank you so much for working tirelessly on it, I wasn't expecting so much dedication!

If I stumble upon any issue, I'll either let you know or raise a PR. Thank you!

**Comment by MikeSchulze** [comment]
Thanks for the sponsoring ;)
